src/inc/msquic.h

```diff
@@ -1473,6 +1473,20 @@ QUIC_STATUS
     _In_ _Pre_defensive_ QUIC_UINT62 ErrorCode // Application defined error code
     );
 
+#ifdef QUIC_API_ENABLE_PREVIEW_FEATURES
+//
+// Shuts the stream down reliably using error code 0x21,
+// which will guarantee the transmission of "ReliableSize" bytes before aborting the stream.
+//
+typedef
+_IRQL_requires_max_(DISPATCH_LEVEL)
+QUIC_STATUS
+(QUIC_API * QUIC_STREAM_SHUTDOWN_RELIABLE_FN)(
```

Comment:
Instead of a new API, let's use a SetParam parameter to configure a reliable reset offset. Then you just use the normal shutdown function.

Reply:
sounds good